ferry/database/diff_db.py

```diff
@@ -0,0 +1,308 @@
+# TODO load data into pandas
+
+import pandas as pd
+import csv
+import json
+import ast
+from io import StringIO
+import logging
+from pathlib import Path
+
+from sqlalchemy import MetaData, text, inspect
+
+from ferry import database
+from ferry.crawler.classes.parse import normalize_unicode
+from ferry.database import Database, Base
+from ferry.transform import transform
+
+
+queries_dir = Path(__file__).parent / "queries"
+
+
+def get_dfs(database_connect_string: str):
+    db = Database(database_connect_string)
+
+    # sorted tables in the database
+    db_meta = MetaData()
+    db_meta.reflect(bind=db.Engine)
+
+    conn = db.Engine.connect()
+    
+    # get table names
+    query = "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'"
+    result = conn.execute(text(query))
+    tables = [row[0] for row in result]
+
+    # Initialize a dictionary to store DataFrames
+    dataframes = {}
+
+    # Load each table into a DataFrame
+    for table_name in tables:
+        df = pd.read_sql_table(table_name, con=conn)
+        dataframes[table_name] = df
+
+    return dataframes
+
+def check_change(row, table_name):
+    cols_to_exclude = {
+        "flags" : [],
+        "course_flags" : [],
+        "professors" : [],
+        "course_professors" : [],
+        'courses' : ['same_course_and_profs_id', 'same_course_id', 'same_prof_id', 'last_offered_course_id'],
+        "listings" : [],
+    }
+
+    for col_name in row.index.tolist():
+        if ("_old" not in col_name):
+            continue
+        col_name = col_name.replace("_old", "")
+        
+        if (col_name in cols_to_exclude[table_name]):
+            return False
+        
+        old_value = row[col_name + "_old"]
+        new_value = row[col_name + "_new"]
+        
+        if isinstance(old_value, list) or isinstance(new_value, list):
+            old_value = ast.literal_eval(str(old_value).replace('"',"'")) # fix quotes
+            new_value = ast.literal_eval(str(new_value).replace('"',"'"))
```

Comment:
I'm pretty sure that strings in DB are JSON, not Python literals, so you can just use `ujson.loads` instead of `ast.literal_eval`